src/server/scripts/EasternKingdoms/zone_dun_morogh.cpp

```diff
@@ -83,7 +96,639 @@ class npc_frozen_mountaineer : public CreatureScript
+                        Troll4GUID = Troll4->GetGUID();
+                        bSummoned = true;
+                    }
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            Creature* agent;
+            if (SummonTimer < diff)
+            {
+
+                if (Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID))
+                {
+                    switch (Phase)
+                    {
+                    case 0:
+                    {
+                        Talk(TALK_0);
+                        SummonTimer = 3500;
+                        Phase++;
+                        break;
+                    }
+                    case 1:
+                    {
+                        agent = me->FindNearestCreature(NPC_AMBERSTILL_MOUNTAINEER, 25.0f, true);
+                        agent->AI()->Talk(TALK_0);
+                        SummonTimer = 4500;
+                        Phase++;
+                    }
+                    case 2:
+                    {
+                        Talk(TALK_1);
+                        SummonTimer = 5500;
+                        Phase++;
+                        break;
+                    }
+                    case 3:
+                    {
+                        Talk(TALK_2);
+                        SummonTimer = 6500;
+                        Phase++;
+                        break;
+                    }
+                    case 4:
+                    {
+                        agent = me->FindNearestCreature(NPC_AMBERSTILL_MOUNTAINEER, 25.0f, true);
+                        agent->AI()->Talk(TALK_1);
+                        SummonTimer = 7500;
+                        Phase++;
+                    }
+                    case 5:
+                    {
+                        Talk(TALK_3);
+                        SummonTimer = 8500;
+                        Phase++;
+                        break;
+                    }
+                    case 6:
+                    {
+                        SummonSpawns();
+                        SummonTimer = 9500;
+                        Phase++;
+                        break;
+                    }
+                    case 7:
+                    {
+                        agent = me->FindNearestCreature(NPC_FROSTMANE_RAIDER, 25.0f, true);
+                        agent->AI()->Talk(TALK_0);
+                        SummonTimer = 10500;
+                        Phase++;
+                    }
+
+                    case 8:
+                    {
+                        Creature* Troll1 = ObjectAccessor::GetCreature(*me, Troll1GUID);
+                        Creature* Troll2 = ObjectAccessor::GetCreature(*me, Troll2GUID);
+                        Creature* Troll3 = ObjectAccessor::GetCreature(*me, Troll3GUID);
+                        Creature* Troll4 = ObjectAccessor::GetCreature(*me, Troll4GUID);
+                        Troll1->SetReactState(REACT_AGGRESSIVE);
```

Comment:
im sure that it'll be easyer with array of Trolls 🗡 
for (auto const& trolGuid : guidsArray)
{
    if (auto const& troll = = ObjectAccessor::GetCreature(*me, trolGuid ))
{
troll ->SetReactState(REACT_AGGRESSIVE);
troll ->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
}

}